### DISCUSSION TITLE
Multithreaded download of a file

### DISCUSSION THREAD

**Discussion by eugrus** [RsyncProject/rsync#512] [Ideas]
Are there any plans for a multithreaded download feature?

**Comment by tmknight** [comment]
xargs and rsync:
https://gist.github.com/MADscientist314/a892e2b50aa7a828f081d88e0d1ff393

**Comment by aguaviva** [comment]
I am curious, What are the advantages of using multithreading? I'd assume that if you use N threads, the connection speed will get divided by N.

Context: I am working on https://github.com/aguaviva/sftp-documents-provider and decided to settle for that reason to just 1 thread. But I am willing to learn! :)
